service/controller/app/resource/releasemigration/create.go

```diff
@@ -262,34 +277,93 @@ func (r *Resource) uncordonChart(ctx context.Context, g8sClient versioned.Interf
 	return nil
 }
 
-func (r *Resource) hasHelmV2ConfigMaps(k8sClient kubernetes.Interface, releaseName, tillerNamespace string) (bool, error) {
+func (r *Resource) hasHelmV2ConfigMaps(k8sClient k8sclient.Interface, tillerNamespace string) (bool, error) {
+	chartMap, err := getChartMap(k8sClient, r.chartNamespace)
+	if err != nil {
+		return false, microerror.Mask(err)
+	}
+
 	lo := metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("%s=%s,%s=%s", "NAME", releaseName, "OWNER", "TILLER"),
+		LabelSelector: fmt.Sprintf("%s=%s", "OWNER", "TILLER"),
 	}
 
 	// Check whether helm 2 release configMaps still exist.
-	cms, err := k8sClient.CoreV1().ConfigMaps(tillerNamespace).List(lo)
+	cms, err := k8sClient.K8sClient().CoreV1().ConfigMaps(tillerNamespace).List(lo)
 	if err != nil {
 		return false, microerror.Mask(err)
 	}
 
-	return len(cms.Items) > 0, nil
+	var count int
+	for _, cm := range cms.Items {
+		if _, ok := chartMap[cm.GetLabels()["NAME"]]; !ok {
+			continue
+		}
+		count++
+	}
+
+	return count > 0, nil
 }
 
-func (r *Resource) hasHelmV3Secrets(k8sClient kubernetes.Interface, releaseName, releaseNamespace string) (bool, error) {
-	lo := metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("%s=%s,%s=%s", "name", releaseName, "owner", "helm"),
+func (r *Resource) hasHelmV3Secrets(k8sClient k8sclient.Interface) (bool, error) {
+	var releaseNamespaces []string
+	{
+		list, err := k8sClient.G8sClient().ApplicationV1alpha1().Charts(r.chartNamespace).List(metav1.ListOptions{})
+		if err != nil {
+			return false, microerror.Mask(err)
+		}
+
+		namespaces := map[string]bool{}
+		for _, chart := range list.Items {
+			namespaces[chart.Spec.Namespace] = true
+		}
+
+		for name, _ := range namespaces {
+			releaseNamespaces = append(releaseNamespaces, name)
```

Comment:
Helm V3 secrets are distributed on whole namespaces, we need to get the secrets from the namespace `charts` CRs specified.